submissions/YuliiaDyka/oop-exercise/index.js

```diff
@@ -0,0 +1,80 @@
+class LivingBeing {
+   constructor(species, name, gender, saying, legs) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.legs = legs;
+   };
+
+   showProperties() {
+      return [
+         this.species, 
+         this.name, 
+         this.gender, 
+         this.saying, 
+         this.legs,
+      ].join('; ');
+   }
+};
+
+class Human extends LivingBeing {
+   constructor(name, gender, saying, legs) {
+      super("Human", name, gender, saying, legs);
+      this.hands = 2;
+   };
+
+   showProperties() {
+      return [
+         super.showProperties(),
+         this.hands,
+      ].join('; ');
+   }
+
+};
+
+class Animal extends LivingBeing {
+   constructor(species, name, gender, saying, legs) {
+      super(species, name, gender, saying, legs);
+   };
+};
+
+class Cat extends Animal {
+   constructor(name, gender, legs) {
+      super("Kitty", name, gender, "Meooow!", legs);
+   };
+};
+
+class Dog extends Animal {
+   constructor(name, gender, legs) {
+      super("Puppy", name, gender, "Woof!", legs);
+   };
+};
+
+class CatWoman extends Cat {
+   constructor(name) {
+      super(name, "Female");
+      this.species = "Cat-woman";
+      this.legs = 2;
+      this.hands = 2;
+   };
+
+   showProperties() {
+      return [
+         super.showProperties(),
+         this.hands,
+      ].join('; ');
+   }
+};
```

Comment:
You may see, how this class breaks OOP, DRY etc
Not your fault.
Here is the learning: OOP is not appropriate for hybrids, non strictly hierarchical relations in the system.
Read about compostion vs inheritance to recognize situations where either is more appropriate.